src/time_unit/days_of_week.rs

```diff
@@ -2,13 +2,22 @@ use crate::error::*;
 use crate::ordinal::{Ordinal, OrdinalSet};
 use crate::time_unit::TimeUnitField;
 use std::borrow::Cow;
+use lazy_static::lazy_static;
+
+lazy_static!{
+    static ref ALL: OrdinalSet = DaysOfWeek::supported_ordinals();
+}
 
 #[derive(Clone, Debug)]
-pub struct DaysOfWeek(OrdinalSet);
+pub struct DaysOfWeek{
+    ordinals: Option<OrdinalSet>
+}
 
 impl TimeUnitField for DaysOfWeek {
-    fn from_ordinal_set(ordinal_set: OrdinalSet) -> Self {
-        DaysOfWeek(ordinal_set)
+    fn from_ordinal_set(ordinal_set: Option<OrdinalSet>) -> Self {
```

Comment:
Accepting an `Option<_>` works fine, but to my eyes `DaysOfWeek::from_ordinal_set(None)` looks like it's using the empty set rather than all values.

Another option here: the `TimeUnitField` trait could have three methods, a low-level one that accepts an `Option<OrdinalSet>` (essentially a rename of the existing `from_ordinal_set`) and two default impls that expose a more self-documenting API for use elsewhere in the crate like `parsing.rs`:

```rust
// Each type implements this:
fn from_optional_ordinal_set(ordinal_set: Option<OrdinalSet>) -> Self;

// === The trait provides these fn impls for free ===

fn all() -> Self {
  Self::from_optional_ordinal_set(None)
}

fn from_ordinal_set(ordinal_set: OrdinalSet) -> Self {
  Self::from_optional_ordinal_set(Some(ordinal_set))
}
```

Reply:
Yeah no problem! I'll do that.